mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+use tokio::process::Command;
+use crate::lvm::error::Error;
+
+use rpc::mayastor::{CreatePoolRequest, Pool, PoolType, PoolState};
+
+const PVCREATE_COMMAND: &str = "pvcreate";
+const PVS_COMMAND: &str = "pvs";
+const PVREMOVE_COMMAND: &str = "pvremove";
+const VGCREATE_COMMAND: &str = "vgcreate";
+const VGS_COMMAND: &str = "vgs";
+const VGREMOVE_COMMAND: &str = "vgremove";
+
+pub async fn lookup_lvm_vg(name: &str) -> Option<Pool> {
+    let pools = list_lvm_vg().await.ok()?;
+    match pools.iter().find(|p| p.name == name) {
+        Some(p) => {
+            return Some(Pool{
```

Comment:
this is fine *but* -- you could write this as `pools.iter().filter(|p| p.name == name) if you where to implement the TryFrom trait and/or Into. Those are just minor things that do not change anything ofcours.